pkg/cache/v2/cache.go

```diff
@@ -51,21 +52,66 @@ type Cache interface {
 	Fetch(context.Context, Request) (*Response, error)
 }
 
+type ResponseType interface {
+	GetMarshalled() ([]*any.Any, error)
+}
+
 // Response is a pre-serialized xDS response.
 type Response struct {
+	ResponseType
 	// Request is the original request.
 	Request discovery.DiscoveryRequest
 
 	// Version of the resources as tracked by the cache for the given type.
 	// Proxy responds with this version as an acknowledgement.
 	Version string
 
-	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
-	ResourceMarshaled bool
-
 	// Resources to be included in the response.
 	Resources []types.Resource
 
+	// The value indicating whether the resource is marshaled, and only one of `Resources` and `MarshaledResources` is available.
+	isResourceMarshaled bool
+
 	// Marshaled Resources to be included in the response.
-	MarshaledResources []types.MarshaledResource
+	marshaledResources []*any.Any
+}
+
+type PassthroughResponse struct {
+	ResponseType
+	// Request is the original request.
+	Request discovery.DiscoveryRequest
+
+	// Version of the resources as tracked by the cache for the given type.
+	// Proxy responds with this version as an acknowledgement.
+	Version string
+
+	// Resources to be included in the response.
+	Resources []*any.Any
```

Comment:
Mentioned in the issue, but how about storing the `DiscoveryResponse` instead?

Reply:
yes, that makes sense, will do